src/components/Filters/FilterModal.tsx

```diff
@@ -1,243 +1,294 @@
+		);
+		if (!!isInvalid) {
+			return;
 		}
-		const searchTermRegEx = new RegExp(searchTerm, 'i');
-		return fieldOptions.filter((option) => {
-			return (
-				option.title.match(searchTermRegEx) ||
-				option.field.match(searchTermRegEx)
-			);
-		});
-	}, [searchTerm, fieldOptions]);
+		const filter = createFilter(schema, signatures as FilterSignature[]);
+		return !!filter ? addFilter(filter) : onClose();
+	};
 
 	return (
 		<Modal
 			title="Filter by"
 			cancel={onClose}
-			done={() => addFilter(filters)}
+			done={createAndAddFilter}
 			action="Save"
 		>
-			{filters.map(({ field, operator, value }, index) => {
-				const operators = SchemaSieve.getOperators(schema, field);
-				return (
-					<RelativeBox key={index}>
-						{index > 0 && <Txt my={2}>OR</Txt>}
-						<Flex>
-							<Box flex={1}>
-								<Select<{ field: string; title: string }>
-									id="filtermodal__fieldselect"
-									options={filteredFieldOptions}
-									onSearch={debouncedSetSearchTerm}
-									searchPlaceholder="Search..."
-									valueKey="field"
-									labelKey="title"
-									// TODO: Remove this logic and pass the primitive value when this is fixed https://github.com/grommet/grommet/issues/3154
-									value={
-										schema.properties
-											? {
-													field,
-													title:
-														(schema.properties[field] as JSONSchema).title ||
-														field,
-											  }
-											: { field }
-									}
-									onChange={({ option }) => setEditField(option.field, index)}
-								/>
-							</Box>
-							{operators.length === 1 && (
-								<Txt py={2} px={3} align="center">
-									{operators[0].label}
-								</Txt>
-							)}
-							{operators.length > 1 && (
-								<Box flex={1} mx={1}>
-									<Select<{ slug: string; label: any }>
-										options={operators}
-										valueKey="slug"
-										labelKey="label"
-										// TODO: Remove this logic and pass the primitive value when this is fixed: https://github.com/grommet/grommet/issues/3154
-										value={operators.find((x) => x.slug === operator)}
-										onChange={({ option }) =>
-											setEditOperator(option.slug, index)
-										}
-									/>
-								</Box>
-							)}
-							<Flex flex={1} alignItems="center">
-								<FilterInput
-									operator={operator}
-									value={value}
-									schema={schema.properties![field] as JSONSchema}
-									onUpdate={(v: any) => setEditValue(v, index)}
-								/>
-							</Flex>
-						</Flex>
+			{signatures.map((signature, index) => (
+				<Flex flexDirection="column" key={(signature?.field || '') + index}>
+					{index > 0 && <Txt my={2}>OR</Txt>}
+					<Flex>
+						<FilterInputs
+							properties={properties || []}
+							signature={signature}
+							schema={schema}
+							onUpdate={(signature) => addSignature(signature, index)}
+						/>
 						{index > 0 && (
-							<DeleteButton
+							<Button
+								color="rgba(0, 0, 0, 0.4)"
```

Comment:
Where is the color from? Is it in our styleguide?

Reply:
that was a custom colour added 4 years ago on that button. It can effectively become `tertiary.main` which is similar.